index.php

```diff
@@ -198,10 +198,25 @@ private function render_footer() {
     print('<p class="text-center">');
 
     if ($this->frontpage['show_visitor_ip']) {
-      if (isset($_SERVER['HTTP_X_FORWARDED_FOR'])) {
-        print('Your IP address: '.htmlentities($_SERVER['HTTP_X_FORWARDED_FOR']).'<br>');
+      if (
+        $this->misc['enable_http_x_forwarded_for'] === true
+        &&
+        isset($_SERVER['HTTP_X_FORWARDED_FOR'])
+      ) {
+        // The user can pass several proxy's, which each one will add its own IP address,
+        //  so we like to take only the first IP address
+        $ips = explode(',', $_SERVER['HTTP_X_FORWARDED_FOR']);
+        $ip = trim($ips[0]);
+        $requester = filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6 | FILTER_FLAG_IPV4);
```

Comment:
There is a `is_valid_ip_address()` function available in `utils` that can be used.